test/test_sandbox_mgmt.py

```diff
@@ -0,0 +1,67 @@
+#!/bin/env python
+
+import os
+import shutil
+import tempfile
+import unittest
+from unittest.mock import patch, PropertyMock
+
+from common.gratia.common import sandbox_mgmt
+
+class SandboxMgmtTests(unittest.TestCase):
+
+    @patch('gratia.common.config.ConfigProxy.get_GratiaExtension', create=True, return_value='test-extension')
+    def test_GenerateFilename(self, mock_config):
+        """GenerateFilename creates a temporary file and returns the path to the file
+        """
+        prefix = 'test-prefix'
+        temp_dir = '/tmp'
+
+        try:
+            with sandbox_mgmt.GenerateFilename(prefix, temp_dir) as filename:
+                self.assertTrue(os.path.exists(filename.name),
+                                f'Failed to create temporary file ({filename.name})')
+                self.assertEqual(temp_dir.rstrip('/'),
+                                 os.path.dirname(filename.name),
+                                 f'Temporary file {filename.name} placed in the wrong directory')
+                self.assertRegex(filename.name,
+                                 rf'{temp_dir}/*{prefix}\.\d+\.{mock_config.return_value}__\w+',
+                                 'Unexpected file name format')
+        finally:
+            try:
+                filename.close()
+                os.remove(filename.name)
+            except (FileNotFoundError, NameError):
+                # don't need to clean up what's not there
+                pass
+
+class CompressOutboxTests(unittest.TestCase):
+    def setUp(self):
+        #provision test environment
+        self.probe_dir = tempfile.mkdtemp()
+        self.outbox = os.path.join(self.probe_dir, 'outbox')
+        os.makedirs(self.outbox, exist_ok=True)
+        self.outfiles = ['testfile1', 'testfile2']
+        # add content to the files
+        for fname in self.outfiles:
+            with open(os.path.join(self.outbox, fname), 'w') as f:
+                f.write('test content')
+
+    def tearDown(self):
+        #Remove probe_dir after test
+        shutil.rmtree(self.probe_dir)
+
+    @patch('gratia.common.config.ConfigProxy.get_GratiaExtension', create=True, return_value ='test-extension')
+    @patch('gratia.common.config.ConfigProxy.getFilenameFragment', create=True, return_value ='test-filename')
+    def test_compress_outbox(self, mock_gratia_ext, mock_file_frag):
+        """CompressOutbox compresses the files in the outbox directory
+            and stores the resulting tarball in probe_dir/staged.
+        """
+        #Parameters for function
+        probe_dir = self.probe_dir
+        outbox = self.outbox
+        outfiles = self.outfiles
+
+        #Assert that CompressOutbox returns True
+        result = sandbox_mgmt.CompressOutbox(probe_dir, outbox, outfiles)
+        self.assertTrue(result)
```

Comment:
A few weeks back we had a discussion about the different things that this function does: returns a boolean but also a tarball gets created in an expected location made up of the specified outfiles. It may be useful to test for things like:

1. The tarball is created in the expected location with expected pattern for the filename (similar to assertions in https://github.com/opensciencegrid/gratia-probe/pull/176/files#diff-a41929032101b1500faa4564484c19a2829fb12c86cd3b95182aa9a66e5071daR22-R29)
2. The unpacked tarball contents are the same as what's given in the outfiles

I'd probably organize these as separate tests and you could even refactor the call to the `CompressOutbox` function into the `setUp()` test!